homeassistant/components/light/xiaomi_miio.py

```diff
@@ -452,8 +553,189 @@ def max_mireds(self):
         """Return the warmest color_temp that this light supports."""
         return 370
 
+    async def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+        try:
+            state = await self.hass.async_add_job(self._light.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._state = state.is_on
+            self._brightness = ceil((255 / 100.0) * state.brightness)
+            self._color_temp = self.translate(
+                state.color_temperature,
+                CCT_MIN, CCT_MAX,
+                self.max_mireds, self.min_mireds)
+
+            delayed_turn_off = self.delayed_turn_off_timestamp(
+                state.delay_off_countdown,
+                dt.utcnow(),
+                self._state_attrs[ATTR_DELAYED_TURN_OFF])
+
+            self._state_attrs.update({
+                ATTR_SCENE: state.scene,
+                ATTR_DELAYED_TURN_OFF: delayed_turn_off,
+                ATTR_NIGHT_LIGHT_MODE: state.smart_night_light,
+                ATTR_AUTOMATIC_COLOR_TEMPERATURE:
+                    state.automatic_color_temperature,
+            })
 
-class XiaomiPhilipsEyecareLamp(XiaomiPhilipsGenericLight, Light):
+        except DeviceException as ex:
+            self._state = None
```

Comment:
Instead of just setting state to None, the available property should return False